File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer,Integer>mpp=new HashMap<>();
        int sum=0;
        mpp.put(sum,1);
        int ans=0;

        for(int x:nums){
            sum+=x;
            ans+=mpp.getOrDefault(sum-k,0);
            mpp.put(sum,mpp.getOrDefault(sum,0)+1);
        }
        return ans;
    }
}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int>mpp;
        int sum=0;
        mpp[0]=1;
        int ans=0;

        for(auto x:nums){
            sum+=x;
            ans+=mpp[sum-k];
            mpp[sum]++;
        }
        return ans;
    }
};